### PR TITLE
Add `proxy-buffer-size` config option

### PR DESCRIPTION
**Pull request by edisile**
### Overview

Added a config option allowing the use of `nginx.ingress.kubernetes.io/proxy-buffer-size` in ingress definition, as detailed in #229

### Rationale
Copy-paste from #229 for the sake of convenience:

> The default ingress configuration creates a 4KB headers proxy buffer, leading to 502 errors when the upstream server's response includes a header section that doesn't fit in the buffer. K8s allows to configure the buffer size, by using the `nginx.ingress.kubernetes.io/proxy-buffer-size` annotation on the ingress definition, allowing the user to select an appropriate value based on their needs.
>
> We are currently facing this issue in our snapcraft.io Juju deployments: the `/login` call triggers a 302 response that carries a large `Location` header for the redirect to Ubuntu One SSO and a `Set-Cookie` header containing session data. When testing on a local environment we can access and modify the ingress definition manually by using `kubectl`, but this workaround isn't suited for a production environment.

### Juju events changes
None

### Module changes
None

### Library changes
None

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation for charmhub is updated
- [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `senior-review-required`)
  - Not sure what the requirements are to apply each of the labels, I'd argue this is a `trivial` change that is also somewhat `urgent`
- [ ] The changelog is updated with changes that affect the users of the charm.
  - there should be no noticeable changes for users

### CODE REVIEW

**Review by srbouffard — APPROVED** [review]

**Review by weiiwang01 — APPROVED** [review]

File: lib/charms/nginx_ingress_integrator/v0/ingress.py
Comment (weiiwang01) [inline lib/charms/nginx_ingress_integrator/v0/ingress.py:1]: You don't need to modify this charm library because `nginx_ingress_integrator.v0.ingress` is already deprecated. No new features will be added.

**Review by swetha1654 — APPROVED** [review]

File: src/ingress_definition.py [hunk 1]
```diff
@@ -286,6 +286,12 @@ def owasp_modsecurity_custom_rules(self) -> str:
             str, self._get_config_or_relation_data("owasp-modsecurity-custom-rules", "")
         ).replace("\\n", "\n")
 
+    @property
+    def proxy_buffer_size(self) -> str:
```
Comment (gregory-schiano) [inline src/ingress_definition.py:290]: It's recommended (as a good practice) to ensure that the proxy buffer size is a multiple of the memory page size. I'm wondering if we shouldn't at least send a warning when the configured value is not (or even enforce it)

Comment (edisile) [reply src/ingress_definition.py:290]: Agreed, I'll add a warning for this case.
Also, I just realized that users could even set the value to 0 which would completely break the proxy, so I'll add some validation for this edge case and trigger an error

**Comment by weiiwang01** [comment]
I have solved the problem in integration tests, it’s the same issue as in wordpress-k8s (pylibjuju version issue). I fixed the problem in another pull request and merged it into main.

@edisile can you merge main into your branch?
